### PR TITLE
Upgrade to 10.3

### LINKED ISSUE #96: Is there a SageMath 10.3 build?

### PR DESCRIPTION
**Pull request by saraedum**
Fixes #96.

##### Dependencies:

* [x] https://github.com/conda-forge/normaliz-feedstock/pull/53
* [x] Upgrade GAP to FLINT 3 (requires normaliz, see https://conda-forge.org/status/migration/libflint30)
* [x] https://github.com/conda-forge/sagelib-feedstock/pull/174

##### Dependency changes

* [x] sagelib 10.2 → 10.3
* [x] alabaster 0.7.12 → 0.7.16
* [x] anyio missing → 4.0.0 (does not seem relevant)
* [x] arrow missing → 1.3.0 (does not seem relevant)
* [x] asttokens 2.1.0 → 2.4.1 (not a dependency currently)
* [x] async_lru missing → 2.0.4 (does not seem relevant)
* [x] babel 2.12.1 → 2.14.0
* [x] beautifulsoup4 4.11.1 → 4.12.2 (not a dependency currently)
* [x] bleach 5.0.1 → 6.1.0
* [x] certifi 2023.7.22 → 2023.11.17 (should be up-to-date already)
* [x] charset_normalizer 2.1.1 → 3.3.2 (not a dependency currently)
* [x] cmake 3.27.3 → 3.27.8 (not a dependency currently)
* [x] comm missing → 0.1.4 (does not seem relevant)
* [x] configure (not relevant for us)
* [x] conway_polynomials 0.5 → 0.8
* [x] cvxopt 1.3.0 → 1.3.2
* [x] cvxpy 1.3.0 → 1.4.1 (not a dependency currently)
* [x] cylp 0.91.5 → 0.92.2
* [x] cython 3.0.4 → 3.0.7 (should be up-to-date)
* [x] database_knotinfo 2023.6.1 → 2024.2.1 (not a dependency currently)
* [x] debugpy 1.6.3 → 1.8.0 (not a dependency currently)
* [x] distlib 0.3.7 → 0.3.8 (not a dependency currently)
* [x] docutils 0.19 → 0.20.1
* [x] e_antic 1.3.0 → 2.0.0 (not a dependency currently)
* [x] ecl 21.2.1 → 23.9.9 (update not needed)
* [x] eclib 20230424 → 20231212 (should be up-to-date)
* [x] exceptiongroup missing → 1.2.0 (does not seem relevant)
* [x] filelock 3.12.3 → 3.13.1 (not a dependency currently)
* [x] flint 2.9.0 → 3.0.1 (need to pull in migration)
* [x] fqdn missing → 1.5.1 (does not seem relevant)
* [x] furo 2022.9.29 → 2023.9.10 (not a dependency currently)
* [x] gmp 6.2.1 → 6.3.0 (already on conda-forge)
* [x] gmpy2 2.1.2 → 2.2.0a1 (update not needed)
* [x] gnumake_tokenpool missing → 0.0.4 (does not seem relevant)
* [x] h11 missing → 0.14.0 (does not seem relevant)
* [x] hatch_vcs 0.3.0 → 0.4.0 (does not seem relevant)
* [x] hatchling 1.18.0 → 1.20.0 (does not seem relevant)
* [x] httpcore missing → 1.0.4 (does not seem relevant)
* [x] httpx missing → 0.27.0 (does not seem relevant)
* [x] idna 3.4 → 3.6 (not a dependency currently)
* [x] igraph 0.10.4 → 0.10.8 (not a dependency currently)
* [x] importlib_metadata 6.8.0 → 7.0.1 (not a dependency currently)
* [x] importlib_resources 6.0.1 → 6.1.1 (not a dependency currently)
* [x] ipykernel 6.6.0 → 6.27.1 (not a dependency currently)
* [x] ipython 8.6.0 → 8.18.1
* [x] ipywidgets 8.0.2 → 8.1.1
* [x] isoduration missing → 20.11.0 (does not seem relevant)
* [x] jedi 0.18.1 → 0.19.1 (not a dependency currently)
* [x] jinja2 3.1.2 → 3.1.3 (the exact pin probably does not matter)
* [x] json5 missing → 0.9.14 (does not seem relevant)
* [x] jsonpointer missing → 2.4 (does not seem relevant)
* [x] jsonschema 4.17.1 → 4.17.3
* [x] jsonschema_specifications missing → 2023.3.3 (does not seem relevant)
* [x] jupyter_client 7.4.4 → 8.3.1
* [x] jupyter_core 4.12.0 → 5.3.2
* [x] jupyter_events missing → 0.6.3 (does not seem relevant)
* [x] jupyter_lsp missing → 2.2.0 (does not seem relevant)
* [x] jupyter_server missing → 2.7.3 (does not seem relevant)
* [x] jupyter_server_terminals missing → 0.4.4 (does not seem relevant)
* [x] jupyterlab missing → 4.1.3 (does not seem relevant)
* [x] jupyterlab_mathjax2 missing → 4.0.0 (does not seem relevant)
* [x] jupyterlab_pygments 0.1.2 → 0.2.2 (does not seem relevant)
* [x] jupyterlab_server missing → 2.24.0 (does not seem relevant)
* [x] jupyterlab_widgets 3.0.3 → 3.0.9 (does not seem relevant)
* [x] markupsafe 2.1.3 → 2.1.4 (not a dependency currently)
* [x] matroid_database missing → 0.2 (does not seem relevant)
* [x] meson 1.2.3 → 1.3.1 (not a dependency currently)
* [x] meson_python 0.14.0 → 0.15.0 (not a dependency currently)
* [x] mpc 1.1.0 → 1.3.1 (the exact pin probably does not matter)
* [x] mpfr 4.0.1.p0 → 4.2.1 (the exact pin probably does not matter)
* [x] mpfrcx 0.5 → 0.6.3 (not a dependency currently)
* [x] nauty 2.8.6.p1 → 2.8.8.p0
* [x] nbclient 0.7.0 → 0.8.0
* [x] nbconvert 7.2.3 → 7.9.2
* [x] nbformat 5.7.0 → 5.9.2
* [x] nest_asyncio 1.5.6 → 1.5.8 (not a dependency currently)
* [x] networkx 3.1 → 3.2.1
* [x] normaliz 3.10.1 → 3.10.1.p0 (not a dependency currently)
* [x] notebook 6.4.12 → 7.1.1
* [x] notebook_shim missing → 0.2.3 (does not seem relevant)
* [x] numpy 1.26.1 → 1.26.3 (the exact pin probably does not matter)
* [x] openblas 0.3.25 → 0.3.26 (the exact pin probably does not matter)
* [x] osqp_python 0.6.2.post8 → 0.6.3 (not a dependency currently)
* [x] overrides missing → 7.4.0 (does not seem relevant)
* [x] pathspec 0.10.2 → 0.12.1 (not a dependency currently)
* [x] pexpect 4.8.0 → 4.9.0
* [x] platformdirs 3.11.0 → 4.1.0 (not a dependency currently)
* [x] polymake 4.9 → 4.11 (not a dependency currently)
* [x] prompt_toolkit 3.0.24 → 3.0.43 (not a dependency currently)
* [x] psutil missing → 5.9.6 (does not seem relevant)
* [x] pygments 2.13.0 → 2.17.2
* [x] pynormaliz 2.18 → 2.18.p0 (not a dependency currently)
* [x] pyrsistent 0.19.2 → 0.19.3 (not a dependency currently)
* [x] python_igraph 0.10.4 → 0.11.3 (not a dependency currently)
* [x] python_json_logger missing → 2.0.7 (does not seem relevant)
* [x] pyyaml missing → 6.0.1 (does not seem relevant)
* [x] qepcad B.1.72 → 1.74 (not a dependency currently)
* [x] referencing missing → 0.23.0 (does not seem relevant)
* [x] requests 2.28.1 → 2.31.0 (does not seem relevant)
* [x] rfc3339_validator missing → 0.1.4 (does not seem relevant)
* [x] rfc3986_validator missing → 0.1.1 (does not seem relevant)
* [x] saclib 2.2.7 → 2.2.8 (not a dependency currently)
* [x] scipy 1.11.3 → 1.11.4
* [x] scs 3.2.2 → 3.2.3 (not a dependency currently)
* [x] setuptools 68.2.2 → 69.0.2 (the exact pin should not matter)
* [x] sniffio missing → 1.3.0 (does not seem relevant)
* [x] soupsieve 2.3.2.post1 → 2.5 (not a dependency currently)
* [x] sphinx 5.2.3 → 7.2.6
* [x] sphinx_basic_ng 0.0.1a12 → 1.0.0b2 (not a dependency currently)
* [x] sphinx_copybutton 0.5.1 → 0.5.2 (not a dependency currently)
* [x] sphinx_inline_tabs missing → 2023.4.21 (does not seem relevant)
* [x] sphinxcontrib_applehelp 1.0.2 → 1.0.8 (not a dependency currently)
* [x] sphinxcontrib_devhelp 1.0.2 → 1.0.6 (not a dependency currently)
* [x] sphinxcontrib_htmlhelp 2.0.0 → 2.0.5 (not a dependency currently)
* [x] sphinxcontrib_qthelp 1.0.3 → 1.0.7 (not a dependency currently)
* [x] sphinxcontrib_serializinghtml 1.1.5 → 1.1.10 (not a dependency currently)
* [x] sphinxcontrib_websupport 1.2.4 → 1.2.7 (not a dependency currently)
* [x] stack_data 0.6.1 → 0.6.3 (not a dependency currently)
* [x] symengine 0.10.1 → 0.11.2 (not a dependency currently)
* [x] symengine_py 0.10.0 → 0.11.0 (not a dependency currently)
* [x] terminado 0.17.0 → 0.17.1
* [x] texttable 1.6.7 → 1.7.0 (not a dependency currently)
* [x] tornado 6.2 → 6.4
* [x] traitlets 5.9.0 → 5.14.0
* [x] trove_classifiers 2023.8.7 → 2023.11.29 (not a dependency currently)
* [x] types_python_dateutil missing → 2.8.19.14 (does not seem relevant)
* [x] typing_extensions 4.7.1 → 4.8.0 (not a dependency currently)
* [x] tzdata 2022.6 → 2023.3 (not a dependency currently)
* [x] uri_template missing → 1.3.0 (does not seem relevant)
* [x] urllib3 1.26.12 → 2.1.0 (not a dependency currently)
* [x] virtualenv 20.24.4 → 20.25.0 (not a dependency currently)
* [x] wcwidth 0.2.5 → 0.2.12
* [x] webcolors missing → 1.13 (does not seem relevant)
* [x] websocket_client missing → 1.6.4 (does not seem relevant)
* [x] wheel 0.41.2 → 0.42.0 (the exact version should not matter)
* [x] widgetsnbextension 4.0.8 → 4.0.9
* [x] zipp 3.11.0 → 3.17.0 (not a dependency currently)

### CODE REVIEW

**Comment by saraedum** [comment]
@conda-forge-admin, please rerender.

File: conda-forge.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 build_platform: {osx_arm64: osx_64}
 conda_forge_output_validation: true
 provider: {linux_aarch64: default}
-build_with_mambabuild: true
+conda_build_tool: conda-build+conda-libmamba-solver
```
[the remark this answers is not in the document]
Comment (saraedum) [reply conda-forge.yml:4]: Is conda-build fast enough now?

[the remark this answers is not in the document]
Comment (isuruf) [reply conda-forge.yml:4]: Yes, conda-build and conda uses libmamba under the hood now by default.

**Comment by saraedum** [comment]
That looks pretty good.

```
2024-05-25T21:50:42.6129927Z sage -t --random-seed=44593750211497097011109367318015638273 src/sage/interfaces/process.pyx  # 1 doctest failed
2024-05-25T21:50:42.6132418Z sage -t --random-seed=44593750211497097011109367318015638273 src/sage/interfaces/singular.py  # 2 doctests failed
2024-05-25T21:50:42.6134618Z sage -t --random-seed=44593750211497097011109367318015638273 src/sage/matrix/matrix_double_dense.pyx  # 1 doctest failed
2024-05-25T21:50:42.6136655Z sage -t --random-seed=44593750211497097011109367318015638273 src/sage/misc/inline_fortran.py  # 3 doctests failed
2024-05-25T21:50:42.6138384Z sage -t --random-seed=44593750211497097011109367318015638273 src/sage/misc/persist.pyx  # 2 doctests failed
2024-05-25T21:50:42.6140306Z sage -t --random-seed=44593750211497097011109367318015638273 src/sage/repl/ipython_extension.py  # 4 doctests failed
```

**Pull request merged**